doc/book/cookbook.md

```diff
@@ -44,3 +44,58 @@ $app->run();
 The above works, because ever `Application` instance is itself middleware, and, more specifically,
 an instance of [Stratigility's `MiddlewarePipe`](https://github.com/zendframework/zend-stratigility/blob/master/doc/book/middleware.md),
 which provides the ability to compose middleware.
+
+## How can I set 404 page?
+
+We can set 404 page with create new Middleware that set 404 status from response object.
+
+Let's create a `NotFound` middleware:
+
+```php
+namespace Application;
+
+use Zend\Expressive\Template\TemplateInterface;
+
+class NotFound
+{
+    public function __invoke($req, $res, $next)
+    {
+        return $next($req, $res->withStatus(404), 'Page Not Found');
+    }
+}
+```
+
+After it, we need to register the middleware into `invokables` type in our service config:
```

Comment:
This is specific to a single container implementation, zend-servicemanager. We should likely either (a) provide setup for multiple containers, or (b) leave container configuration as an exercise for the reader.